qcfractal/dashboard/dash_models.py

```diff
@@ -0,0 +1,106 @@
+import dash_bootstrap_components as dbc
+import dash_core_components as dcc
+import dash_html_components as html
+from dash.dependencies import Input, Output
+import pandas as pd
+import plotly.graph_objs as go
+from plotly.colors import DEFAULT_PLOTLY_COLORS
+
+import dash_coreui_components as coreui
+from flask import current_app
+
+from .connection import get_socket
+
+_default_margin = {"t": 5, "b": 5, "r": 5, "l": 5}
+
+
+def list_managers(status=None, modified_after=None):
+    socket = get_socket()
+
+    managers = socket.get_managers(status=status, modified_after=modified_after)
+    cols = [
+        "cluster", "username", "tag", "completed", "submitted", "failures", "returned", "created_on", "modified_on",
+        "programs", "procedures",
+         # "name"
+    ]
+
+    df = pd.DataFrame(managers["data"])
+    df = df[cols]
+    df["programs"] = df["programs"].apply(lambda x : ", ".join(sorted(x)))
+    df["procedures"] = df["procedures"].apply(lambda x : ", ".join(sorted(x)))
+    df.columns = [x.title() for x in cols]
+    df.sort_values('Completed', inplace=True, ascending=False)
+    return dbc.Table.from_dataframe(df, style={"width": "100%"})
+
+
+def manager_graph(status=None, modified_after=None):
+    socket = get_socket()
+
+    managers = socket.get_managers(status=status, modified_after=modified_after)
+    df = pd.DataFrame(managers["data"])
+
+    bars = []
+    if df.shape[0] > 0:
+        data = df.groupby("cluster")[["completed", "submitted", "failures"]].sum()
+        data["error"] = data["failures"]
+        data["running"] = data["submitted"] - data["completed"]
+
+        bar_iter = [("error", DEFAULT_PLOTLY_COLORS[3]), ("running", DEFAULT_PLOTLY_COLORS[2]),
+                    ("completed", DEFAULT_PLOTLY_COLORS[0])]
+
+        data.sort_values("completed", inplace=True, ascending=False)
+        bars = []
+        for status, color in bar_iter:
+            bars.append(go.Bar(name=status.title(), x=data.index, y=data[status], marker_color=color))
+
+    return go.Figure(
+        data=bars,
+        layout={
+            # "yaxis_type": "log",
+            "barmode": "stack",
+            "yaxis": {
+                "title": "Completed Tasks"
+            },
+            "xaxis": {
+                "title": "Cluster"
```

Comment:
"Manager" on the x-axis?

Reply:
This is actually `Cluster` since we group on that variable.